bundles/specmate-emfrest/src/com/specmate/emfrest/internal/rest/SpecmateResource.java

```diff
@@ -96,6 +108,31 @@ public final Object delete(@PathParam(SERVICE_KEY) String serviceName) {
 				true);
 	}
 
+	private boolean checkProject(EObject update, boolean recurse) {

```

Comment:
Maybe it would be good to rename the method to something more descriptive, like isProjectModificationRequestAuthorized (or something more concise) and then add a comment how we check authorization (basically, by ensuring that the project from which the modification request origins corresponds to the to be changed project). Also, add a brief explanation why the recursion is necessary.